Mathlib/Tactic/Order/ToInt.lean

```diff
@@ -0,0 +1,187 @@
+/-
+Copyright (c) 2025 Vasilii Nesterov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Vasilii Nesterov
+-/
+import Mathlib.Tactic.Order.CollectFacts
+import Batteries.Data.List.Pairwise
+import Mathlib.Tactic.GeneralizeProofs
+
+/-!
+# Translating linear orders to ℤ
+
+In this file we implement the translation of a problem in any linearly ordered type to a problem in
+`ℤ`. This allows us to use the `omega` tactic to solve it.
+
+While the core algorithm of the `order` tactic is complete for the theory of linear orders in the
+signature (`<`, `≤`),
+it becomes incomplete in the signature with lattices operations `⊓` and `⊔`. With this operations
+the problem becomes NP-hard, and the idea is to reuse some smart and efficient procedure, such as
+`omega`.
+
+## TODO
+
+Migrate to `grind` when it is ready.
+-/
+
+namespace Mathlib.Tactic.Order.ToInt
+
+variable {α : Type*} [LinearOrder α] {n : ℕ} (val : Fin n → α)
+
+theorem exists_translation : ∃ tr : Fin n → ℤ, ∀ i j, val i ≤ val j ↔ tr i ≤ tr j := by
```

Comment:
This is a cool approach! Never expected to see this kind of abstract existence proof being used in a tactic. :D

Reply:
```suggestion
/-- The main theorem asserting the existence of a translation.

We use `Classical.chooose` to turn this into a value for use in the `order` tactic,
see `toInt`.
-/
theorem exists_translation : ∃ tr : Fin n → ℤ, ∀ i j, val i ≤ val j ↔ tr i ≤ tr j := by
```